src/interfaces/hapi.ts

```diff
@@ -79,30 +79,46 @@ export function makeHapiPlugin(client: RequestHandler, config: Configuration) {
    * @returns {Undefined} - returns the execution of the next callback
    */
   function hapiRegisterFunction(server: {}, options: {}, next: Function) {
-    if (isObject(server)) {
-      if (isFunction((server as hapi.Server).on)) {
-        (server as hapi.Server).on('request-error', (req, err) => {
-          client.sendError(hapiErrorHandler(req, err, config));
+    if (server) {
+      if ((server as any).events && (server as any).events.on) {
+        // Hapi 17 is being used
+        (server as any).events.on('log', function(event, tags) {
+          if (event.error) {
+            client.sendError(hapiErrorHandler(undefined, event.error));
+          }
+        });
+
+        (server as any).events.on('request', (request, event, tags) => {
+          if (event.error) {
+            client.sendError(hapiErrorHandler(request, event.error));
+          }
         });
       }
+      else {
+        if (isFunction((server as hapi.Server).on)) {
+          (server as hapi.Server).on('request-error', (req, err) => {
+            client.sendError(hapiErrorHandler(req, err, config));
+          });
+        }
 
-      if (isFunction((server as hapi.Server).ext)) {
-        (server as hapi.Server).ext('onPreResponse', (request, reply) => {
-          if (isObject(request) && isObject(request.response) &&
-              // TODO: Handle the case when `request.response` is null
-              request.response!.isBoom) {
-            const em = hapiErrorHandler(
-                request,
+        if (isFunction((server as hapi.Server).ext)) {
+          (server as hapi.Server).ext('onPreResponse', (request, reply) => {
+            if (isObject(request) && isObject(request.response) &&
```

Comment:
Same question. Can't this be just `if (request && request.response && request.response.isBoom)`?

Reply:
This is existing code that I don't want to change as part of this PR.

Reply:
This is existing code that I don't want to change as part of this PR.  The only thing that should have changed is the indentation level for that block of code.